src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,115 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value !== Extrapolate.EXTEND &&
+    value !== Extrapolate.CLAMP &&
+    value !== Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (
+    (typeof type === 'object' &&
+      (!type.extrapolateLeft &&
+        !type.extrapolateRight &&
+        Object.keys(type).length >= 1 &&
+        type.constructor === Object)) ||
+    (Object.keys(type).length > 2 && type.constructor === Object)
+  ) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      })`
+    );
+  }
+
+  if (
+    typeof type === 'object' &&
+    type.extrapolateLeft &&
+    isExtrapolate(type.extrapolateLeft)
```

Comment:
Won't `isExtrapolate` return `false` for `null` / `undefined`?

Reply:
Yeap it will